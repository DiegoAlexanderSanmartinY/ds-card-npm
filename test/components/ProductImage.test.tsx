import React from 'react';
import { ProductImage, ProductCard } from '../../src/components';
import { product2 } from '../data/products';
import renderer from 'react-test-renderer';

describe('ProductImage', () => {
  test('debe de mostrar el componente correctamente con la imagen personalizada', () => {
    const wrapper = renderer.create(
      <ProductImage
        img="https://www.cdc.gov/coronavirus/2019-ncov/images/symptoms-testing/pcr-test.png?_=15023"
        className="custom-class"
      />
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  test('debe de mostrar el componente con la imagen del producto', () => {
    const wrapper = renderer.create(
      <ProductCard product={product2}>{() => <ProductImage />}</ProductCard>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
