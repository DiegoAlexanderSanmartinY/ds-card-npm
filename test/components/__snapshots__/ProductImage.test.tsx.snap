// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage debe de mostrar el componente con la imagen del producto 1`] = `
<div
  className="undefined productCard"
>
  <img
    alt="coffee"
    className="undefined productImg"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`ProductImage debe de mostrar el componente correctamente con la imagen personalizada 1`] = `
<img
  alt="coffee"
  className="custom-class productImg"
  src="https://www.cdc.gov/coronavirus/2019-ncov/images/symptoms-testing/pcr-test.png?_=15023"
/>
`;
